targets/wasm_exec.js

```diff
@@ -298,9 +298,20 @@
 
 					// func finalizeRef(v ref)
 					"syscall/js.finalizeRef": (v_ref) => {
-						// Note: TinyGo does not support finalizers so this should never be
-						// called.
-						console.error('syscall/js.finalizeRef not implemented');
+						// Note: TinyGo does not support finalizers so this is only called
+						// for one specific case, by js.go:jsString. and can/might leak memory.
+						const id = mem().getUint32(unboxValue(v_ref), true);
+						// Note that this if is so far seemingly never true. Someone should investigate why.
+						// (ie we are silently, no log even, ignoring an unexpected case)
+						if (this._goRefCounts?.[id] !== undefined) {
+							this._goRefCounts[id]--;
+							if (this._goRefCounts[id] === 0) {
+								const v = this._values[id];
+								this._values[id] = null;
+								this._ids.delete(v);
+								this._idPool.push(id);
+							}
+						}
```

Comment:
```suggestion
						} else {
						     console.error("syscall/js.finalizeRef invalid reference count");
					        }
```

Reply:
So something like this is basically what you were proposing to catch the error you referred to? (but with the correct formatting)

Reply:
Sorry about all the edits, GH did not like my spaces.

Reply:
? that's how it was in the first commit ... https://github.com/tinygo-org/tinygo/pull/4352/commits/e4e6e0d32c02e21ebdcba026ed96be7e550b8a54#diff-3c36e56a725ebf4b2f5fd4a5a117e50894d15570f979b76d0affc410430ccec5R314

Reply:
That first commit added two lines with calls to `console.log()`, one in each clause of the `if`. So not quite the same.

Reply:
so you want my original commit, but without the first log?

Reply:
done, PTAL

Reply:
I missed btw you had put `error` because the whole point is not not have `error` (title and description of this PR) thus I didn't get your point (not to say I agree with it, I don't, but my bad on autocorrecting your .error to .log)